File: app/src/main/java/dev/n3shemmy3/coffre/ui/MainActivity.java
package dev.n3shemmy3.coffre.ui;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.splashscreen.SplashScreen;

import dev.n3shemmy3.coffre.R;
import dev.n3shemmy3.coffre.ui.screen.setup.StartScreen;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SplashScreen.installSplashScreen(this);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        if (savedInstanceState == null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.fragmentContainer, new StartScreen())
                    .commit();
        }
    }
}

File: app/src/main/kotlin/dev/n3shemmy3/coffre/MainActivity.kt
package dev.n3shemmy3.coffre

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.runtime.remember
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import androidx.navigation3.runtime.entryProvider
import androidx.navigation3.runtime.rememberNavBackStack
import androidx.navigation3.ui.NavDisplay
import dev.n3shemmy3.coffre.compose.navigation.AppRoute
import dev.n3shemmy3.coffre.compose.screen.detail.DetailScreen
import dev.n3shemmy3.coffre.compose.screen.main.MainScreen
import dev.n3shemmy3.coffre.compose.screen.main.MainViewModel
import dev.n3shemmy3.coffre.compose.screen.search.SearchScreen
import dev.n3shemmy3.coffre.compose.theme.AppTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        installSplashScreen()
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        setContent {
            val backStack = rememberNavBackStack(AppRoute.Main)

            val viewModel = remember {
                MainViewModel(App.appDatabase)
            }

            AppTheme {
                NavDisplay(
                    backStack = backStack,
                    onBack = { backStack.removeLastOrNull() },
                    entryProvider = entryProvider {
                        entry<AppRoute.Main> {
                            MainScreen(backStack, viewModel)
                        }
                        entry<AppRoute.Detail> {
                            DetailScreen(backStack, viewModel)
                        }
                        entry<AppRoute.Search> {
                            SearchScreen(backStack, viewModel)
                        }
                        entry<AppRoute.Settings> {

                        }
                    },
                )
            }
        }
    }
}
